### ISSUE TITLE
Incorrect ZeroPadding before MaxPool2D in keras' resnet

### ISSUE THREAD

**Issue by dlopr** [tensorflow/tensorflow#48105]
Hi,

I've noticed that the implementation of resnet networks in keras introduces a ZeroPadding layer before the initial MaxPool2D 3x3 stride 2:

https://github.com/tensorflow/tensorflow/blob/204082475214b3f08d1301998780592b53076951/tensorflow/python/keras/applications/resnet.py#L161

I don't think this is correct. Zero is not a neutral element for a MaxPool2D operation. The input values at the edges could be negative.

I believe the intention of that zero padding layer is what would be correctly represented as SAME padding for the MaxPool2D directly. Note that the padding layer is also adding padding elements to the right and bottom edges of the input that the 3x3 stride 2 MaxPool2D operation won't ever use if the input is even size as I believe it commonly is.

**Comment by xonx4l** [comment]
Hey! here's a solution I think that can be implemented .

**Comment by xonx4l** [comment]
-: 78ce835ebb152febf280cfc0fa46de791de57eb1

**Comment by xonx4l** [comment]
Here's a pr in eras repo as i am unable to find keras/application on the tensorflow repo . 

-:Update resnet.py #1

**Comment by xonx4l** [comment]
In the PR-I've changed the pool1_pad layer's padding to (1, 0) for the right and bottom edges. This will ensure that the padding is added only on the top and left sides, which compensates for the even-sized input.
I've changed the MaxPooling2D layer's padding to 'valid', which is equivalent to "SAME" padding in this context because of the additional padding added by pool1_pad.

**Comment by dlopr** [comment]
I can't find the pr you're referring to.

In any case, if I understood you correctly, removing the right and bottom padding would only avoid wasteful addition of elements that won't be used.

The main issue was that a value of zero is not correct for padding the input of a maxpool layer with signed type. It should be padded with the minimum representable value of the type or just use same padding and avoid the pad layer if supported.

**Comment by MarkDaoust** [comment]
Hi, the source for the keras code is now github.com/keras-team/keras and soon github.com/keras-team/keras-core.

Actually `keras.applications` is in maintenance mode. users should prefer the implementations in github.com/keras-team/keras-cv. 

Either way, if this issue still exists, please re-raise it on the Keras repo(s).
